File: swift/grade-school/Sources/GradeSchool/GradeSchool.swift
class GradeSchool {
  var roster = [Int: [String]]()
  var sortedRoster: [Int: [String]] {
    return roster.mapValues { $0.sorted() }
  }

  func addStudent(_ name: String, grade: Int) {
    roster[grade] == nil ? roster[grade] = [name] : roster[grade]!.append(name)
  }

  func studentsInGrade(_ grade: Int) -> [String]? {
    return roster[grade] ?? []
  }
}

File: csharp/grade-school/GradeSchool.cs
using System.Collections.Generic;
using System.Linq;

public class GradeSchool
{
    private class Student {
        public string Name { get; set; }
        public int Grade { get; set; }
    }
    
    private List<Student> roster = new List<Student>();

    public void Add(string student, int grade)
    {
        roster.Add(new Student(){
            Name = student,
            Grade = grade
        });
    }

    public IEnumerable<string> Roster() => roster
        .OrderBy(student => student.Grade)
        .ThenBy(student => student.Name)
        .Select(student => student.Name);

    public IEnumerable<string> Grade(int grade) => roster
        .Where(student => student.Grade == grade)
        .OrderBy(student => student.Name)
        .Select(student => student.Name);
}